admin/src/pages/ManageUserRoles/ManageUserRoles.js

```
@@ -0,0 +1,376 @@
+        if (placeRoles.length) {
+            // GET THE IDS AND REPLACE THEM WITH THE ACTUAL OBJECTS.
+            ids = getUniqueIDs(placeRoles, `${place}_id`);
+            let places = await restClient(GET_MANY, `${place}s`, {
+                ids: ids
+            });
+            places = makeIDMapping(places.data);
+
+            placeRoles = placeRoles.reduce((obj, placeRole) => {
+                obj[`${placeRole[`${place}_id`]}:${placeRole.role_id}`] = {
+                    [place]: places[placeRole[`${place}_id`]],
+                    role: roles[placeRole.role_id]
+                };
+                return obj;
+            }, {});
+        }
+        return placeRoles;
+    }
+
+    async handleSelect(rows) {
+        if (rows.length > 0) {
+            try {
+                let roles = this.state.rolesMapping;
+                if (!roles) {
+                    roles = await restClient(GET_MANY, 'roles', {});
+                    roles = makeIDMapping(roles.data);
+                }
+                const user = this.state.userResults[rows[0]];
+                // GET USERDOMAINROLES FOR SELECTED USER.
+                const domainRoles = await this.getUserPlaceRoles(user, 'domain', roles);
+                // GET USERSITEROLES FOR SELECTED USER.
+                const siteRoles = await this.getUserPlaceRoles(user, 'site', roles);
+                // SET THE STATE WITH THE SELECTED USER AND ALL ROLES FOUND.
+                this.setState({
+                    selectedUser: rows[0],
+                    userRoles: {
+                        domainRoles: domainRoles,
+                        siteRoles: siteRoles
+                    }
+                });
+            } catch (error) {
+                this.handleAPIError(error);
+            }
+        }
+    }
+
+    handleDelete(data) {
+        const { userResults, selectedUser, userRoles } = this.state;
+        const user = selectedUser >= 0 ? userResults[selectedUser] : null;
```

wouldn't you want to throw an error or stop the execution of this function if the `selecteUser >=0` ?  instead its continuing and it will throw and api error right?